prov/efa/src/rxr/rxr_pkt_cmd.c

```diff
@@ -84,14 +78,7 @@ ssize_t rxr_pkt_post_data(struct rxr_ep *rxr_ep,
 	 */
 	data_pkt->hdr.seg_offset = tx_entry->bytes_sent;
 
-	/*
-	 * TODO: Check to see if underlying device can support CUDA
-	 * registrations and fallback to rxr_ep_send_data_pkt_entry() if it does
-	 * not. This should be done at init time with a CUDA reg-and-fail flag.
-	 * For now, always send CUDA buffers through
-	 * rxr_pkt_send_data_desc().
-	 */
-	if (efa_is_cache_available(efa_domain) || efa_ep_is_cuda_mr(tx_entry->desc[0]))
+	if (tx_entry->desc[0])
```

Comment:
Don't we still need to enter here during the cache path?

Reply:
No. `rxr_ep_tx_init_mr_desc` sets `tx_entry->desc` after doing the inline registration. The only cache check should be when we decide to do inline registration or not and no where else in the send path. I haven't been able to make those changes yet.

Reply:
We already call ```rxr_ep_prepare_desc_send``` in  ```rxr_pkt_handle_long_rtm_sent```, which will set ```tx_entry->desc[0]``` for the cache path.